python/pyspark/conf.py

```diff
@@ -149,35 +161,53 @@ def setAll(self, pairs):
         :param pairs: list of key-value pairs to set
         """
         for (k, v) in pairs:
-            self._jconf.set(k, v)
+            self.set(k, v)
         return self
 
     def get(self, key, defaultValue=None):
         """Get the configured value for some key, or return a default otherwise."""
         if defaultValue is None:   # Py4J doesn't call the right get() if we pass None
-            if not self._jconf.contains(key):
-                return None
-            return self._jconf.get(key)
+            if self._jconf:
+                if not self._jconf.contains(key):
+                    return None
+                return self._jconf.get(key)
+            else:
+                if key not in self._conf:
+                    return None
+                return self._conf[key]
         else:
-            return self._jconf.get(key, defaultValue)
+            if self._jconf:
+                return self._jconf.get(key, defaultValue)
+            else:
+                return self._conf.get(key, defaultValue)
 
     def getAll(self):
         """Get all values as a list of key-value pairs."""
         pairs = []
-        for elem in self._jconf.getAll():
-            pairs.append((elem._1(), elem._2()))
+        if self._jconf:
+            for elem in self._jconf.getAll():
+                pairs.append((elem._1(), elem._2()))
+        else:
+            for k, v in self._conf.items():
+                pairs.append((k, v))
```

Comment:
This could be simplified:

```python
if self._jconf is not None:
    return [(elem._1(), elem._2()) for elem in self._jconf.getAll()]
else:
    return self._conf.items()